EcoSoapBank/Network Controller/GraphQLController.swift

```diff
@@ -120,92 +148,10 @@ class GraphQLController {
-        let variables: V
-
-        func encode(to encoder: Encoder) throws {
-            var container = encoder.container(keyedBy: CodingKeys.self)
-            try container.encode(query, forKey: .query)
-            var variablesContainer = container.nestedContainer(keyedBy: CodingKeys.self, forKey: .variables)
-            try variablesContainer.encode(variables, forKey: .input)
-        }
-
-    }
-
-    private enum CodingKeys: String, CodingKey {
-        case query
-        case variables
-        case input
-    }
-}
-    
-enum GraphQLError: Error {
-    case noData
-    case invalidData
-    case noToken
-    case unimplemented
-    case backendMessages([String])
 }
 
-/// Protocol to set conformance to possible input types for GraphQL query and mutation variables
-protocol VariableType: Encodable {}
-
-//extension Dictionary: VariableType where Key == GraphQLController.InputTypes, Value == String {}
-extension Dictionary: VariableType where Key == String, Value == String {}
-extension Pickup.ScheduleInput: VariableType {}
-
 // MARK: - Data Providers
 
-extension GraphQLController: UserDataProvider {
-    func logIn(_ completion: @escaping ResultHandler<User>) {
-        guard let token = self.token else {
-            return completion(.failure(GraphQLError.noToken))
-        }
-        queryRequest(User.self,
-                     query: GraphQLMutations.login,
-                     variables: ["token": token],
-                     completion: completion)
-    }
-}
-
-extension GraphQLController: ImpactDataProvider {
-    func fetchImpactStats(
-        forPropertyID propertyID: String,
-        _ completion: @escaping ResultHandler<ImpactStats>
-    ) {
-        // TODO: may need to add token later
-        queryRequest(ImpactStats.self,
-                     query: GraphQLQueries.impactStatsByPropertyId,
-                     variables: ["propertyId": propertyID],
-                     completion: completion)
-    }
-}
-
-extension GraphQLController: PickupDataProvider {
-    func fetchPickups(
-        forPropertyID propertyID: String,
-        _ completion: @escaping ResultHandler<[Pickup]>
-    ) {
-        // TODO: may need to add token later
-        queryRequest([Pickup].self,
-                     query: GraphQLQueries.pickupsByPropertyId,
-                     variables: ["propertyId": propertyID],
-                     completion: completion)
-    }
-
-    func schedulePickup(
-        _ pickupInput: Pickup.ScheduleInput,
-        completion: @escaping ResultHandler<Pickup.ScheduleResult>
-    ) {
-        // TODO: may need to add token later
-        queryRequest(Pickup.ScheduleResult.self,
-                     query: GraphQLQueries.impactStatsByPropertyId,
-                     variables: pickupInput,
-                     completion: completion)
-    }
-}
+extension GraphQLController: UserDataProvider {}
+extension GraphQLController: ImpactDataProvider {}
+extension GraphQLController: PickupDataProvider {}
```

Comment:
I'm guessing you put these methods up above so all the "public" methods would be in one spot? I think personally I prefer having the protocol methods in the extensions where they're declared, but it's not a huge deal of course. 😄